docs/src/components/VisualList.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from "react";
+import styled from "styled-components";
+import {
+  ButtonLink,
+  Card,
+  CardSection,
+  Coupon,
+  Heading,
+  InputField,
+  Stack,
+  Text,
+  Tile,
+} from "@kiwicom/orbit-components";
+import { Link as LinkIcon, Search } from "@kiwicom/orbit-components/icons";
+
+import { IllustrationObjectShape } from "./IllustrationList";
+import { IconObjectShape } from "./IconList";
+import { CodeBlock } from "./Code";
+import { copyTimeout } from "../utils/common";
+
+const Grid = styled.div`
+  display: grid;
+  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
+  grid-gap: ${({ theme }) => theme.orbit.spaceXSmall};
+`;
+
+const StyledAnchorWrapper = styled.div`
+  svg {
+    visibility: hidden;
+  }
+
+  &:hover svg {
+    visibility: visible;
+  }
+`;
+
+type VisualObjectShape = IllustrationObjectShape | IconObjectShape;
+
+interface ObjectActions {
+  actions(
+    VisualObjectShape,
+    copied?: boolean,
+    setCopied?: React.Dispatch<React.SetStateAction<boolean>>,
+  ): React.ReactNode;
+  getImgSource(VisualObjectShape): string;
+  exampleCode(string): string;
+}
+
+interface VisualProps extends VisualListProps {
+  visualName: string;
+  opened: boolean;
+  setOpenedVisual: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const Visual = ({
+  actions,
+  exampleCode,
+  getImgSource,
+  list,
+  opened,
+  setOpenedVisual,
+  visualName,
+}: VisualProps) => {
+  const [copied, setCopied] = useState(false);
+  useEffect(() => {
+    copyTimeout(copied, setCopied);
+  }, [copied, setCopied]);
+
+  const visualObject: VisualObjectShape = list[visualName];
+
+  return (
+    <div id={visualName}>
+      <Tile onClick={() => setOpenedVisual(opened ? "" : visualName)}>
+        <Stack justify="between">
+          <img src={getImgSource(visualObject)} alt="" />
```

Comment:
@CollierCZ I would create small styled components and attach some styles, so we are able to resolve the problematic case with TIcketFlexi illustration. It will at least work for now, until we will have different solution with this exceptional illustration.

```jsx
const StyledImage = styled.img`
  max-height: 180px;
  max-width: 100%;
`
```

Add the same time I would apply `justify="center"` on the wrapping Stack.

Reply:
My suggestion to use [aspect ratio box](https://css-tricks.com/aspect-ratio-boxes/):

```jsx
const StyledRatioContainer = styled.div`
  position: relative;
  padding-bottom: calc(400 / 677 * 100%); /* height / width = ratio */
`
const StyledImage = styled.img`
  position: absolute;
  width: 100%;
  height: 100%;
  object-fit: contain;
`

// ...

<StyledRatioContainer>
  <StyledImage src={getImgSource(visualObject)} alt="" />
</StyledRatioContainer>
```